### ISSUE TITLE
ELM water balance errors with PGI on Summit 

### ISSUE THREAD

**Issue by minxu74** [E3SM-Project/E3SM#4479]
When use the new PGI version 20.1 and 20.4 on Summit, there are ICE errors as follows:
```
PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (5646,base=5649) member symbol shr_infnan_nan$ac (offset=61): not found!  (/autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8/components/elm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (5646,base=5649) member symbol shr_infnan_snan$ac (offset=63): not found!  (/autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8/components/elm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (5646,base=5649) member symbol shr_infnan_qnan$ac (offset=65): not found!  (/autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8/components/elm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
PGF90-S-0000-Internal compiler error. import: module bshr_infnan_mod (5646,base=5649) member symbol shr_infnan_inf$ac (offset=67): not found!  (/autofs/nccs-svm1_home1/mxu/MyGit/e3sm_RHEL8/components/elm/src/external_models/sbetr/src/driver/alm/ALMbetrNLMod.F90: 28)
``` 

It seemed that there were some conflicts between bshr_infnan_mod and shr_infnan_mod.  

I will send an email to OLCF to report the ICEs and meanwhile I will send a request to keep the old PGI 19.9.

**Comment by minxu74** [comment]
As PGI was re-branded to be the NVIDIA HPC Compiler, I used the nvhpc/21.3 and ran the e3sm_developer test suite on Summit.  **45** tests in the suite were built successfully and only 2 tests failed. But only 8 of 45 tests were run successfully, others failed in different reasons.  The passed tests are:
```
    PASS ERS.MOS_USRDAT.RMOSGPCC.summit_pgi.mosart-mos_usrdat RUN time=1044
    PASS ERS.f09_g16_g.MALISIA.summit_pgi RUN time=18
    PASS ERS.f19_g16_rx1.A.summit_pgi RUN time=71
    PASS ERS_Ld5.T62_oQU120.CMPASO-NYF.summit_pgi RUN time=233
    PASS SEQ.f19_g16.X.summit_pgi RUN time=314
    PASS SMS.MOS_USRDAT.RMOSGPCC.summit_pgi.mosart-unstructure RUN time=370
    PASS SMS.ne30_f19_g16_rx1.A.summit_pgi RUN time=31
    PASS SMS_B.ne4_ne4.F-EAM-AQP1.summit_pgi.eam-hommexx RUN
```

All tests with ELM failed with the water imbalance error (larger than 1.e-4 mm).

**Comment by amametjanov** [comment]
@peterdschwartz just curious which PGI/nvhpc version was used in ELM+OpenACC runs?

**Comment by peterdschwartz** [comment]
@amametjanov Originally I believe it was PGI 19.9 on Summit.  Also, I am just now rebasing for more full e3sm runs and using nvhpc 20.9 rather than 21.3  to get rid of ICE for some `acc routine seq` translations.

**Comment by amametjanov** [comment]
Fixed after the merge of #4863.
